### ISSUE TITLE
HTTP expiry on the cookie doesn't match the encrypted/signed client-sessions expiry

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#2754]
tl;dr - main site should more gracefully handle /wsapi/logout 403, or check with /wsapi/session_context first

STR

1) Edit config/local.json and add these two values:

```
"ephemeral_session_duration_ms": 120000,
"authentication_duration_ms": 180000,
```

2) Load http://127.0.0.1:10002 and signup as a brand new user (or use an existing user, but use a clean profile so there is no previous state).

3) When you sign in, a session cookie is issued with embedded values of "now" and duration of 120000, with Expires: set to a date 3 minutes from "now".

4) Wait two minutes, and click 'Sign Out'. The request to http://127.0.0.1:10002/wsapi/logout sends the cookie which is http-unexpired, but expired for client-sessions purposes. The response is 403 with body 'Forbidden: no cookie'. The UI shows an error screen with 'Persona requires cookies to remember you'. 

Alternate STR for the "one month" duration:

1) Do 1,2,3 above and then instead do this:

2) wait 60 seconds after you've signed in, load http://127.0.0.1:10001/ and click 'Get assertion'. When asked, extend the session for "one month". A session cookie returned by prolong_session embedded values of "now" and duration of 180000, with Expires: set to a date 3 minutes from "now".

3) So now, wait three minutes for the client-session duration to expire, then click 'Sign Out'. This time the request to http://127.0.0.1:10002/wsapi/logout sends no cookie, since it is http-expired, but the response and UI is the same (403, with 'Persona requires cookies to remember you').

Note: this doesn't apply for /sign_in dialog, which first call /wsapi/session_context which returns "authenticated":false. Um, I think.

**Comment by lloyd** [comment]
solving on the frontend seems to be right,. just ignore 403s when logout fails

**Comment by jrgm** [comment]
Hmm, I still had this tab open, and just re-read my original report. The key point on this issue is:

```
The request to http://127.0.0.1:10002/wsapi/logout sends the cookie which is
http-unexpired, but expired for client-sessions purposes. The response is 403
with body 'Forbidden: no cookie'. The UI shows an error screen with 'Persona
requires cookies to remember you'.
```

So, I haven't looked through the code but there's a mis-diagosis presented by
the UI. It's not that the user does not have cookies enabled, it's that the
client-sessions cookie has expired.

So, yes, the specific case noted in this issue can be solved for logout
on main site, but I'm wondering if this is more general (given occasional
reports of "needs cookies" by users who know they are enabled in their UA).

**Comment by fmarier** [comment]
I have just prepared a pull request related to this on client-sessions itself: https://github.com/mozilla/node-client-sessions/pull/55

**Comment by shane-tomlinson** [comment]
@fmarier - is the updated node-client-sessions being used in Persona now?

**Comment by fmarier** [comment]
@shane-tomlinson Not yet. I'm not done with my patch on browserid itself: https://github.com/fmarier/browserid/commits/ephemeral_cookie (the ephemeral case works, but the non-ephemeral case doesn't work).

**Comment by callahad** [comment]
Hi! To help us better focus, I'm "closing" all issues that have been open for more than six months. These have been tagged "cleanup-2014" so that we can go back and review them in the future.

For more information, check out this thread: http://thread.gmane.org/gmane.comp.mozilla.identity.devel/7394

If you believe this bug is still a major issue for you, please comment, submit a pull request, or discuss it on our mailing list: https://lists.mozilla.org/listinfo/dev-identity

Sorry for GitHub notification spam!
